src/signal.rs

```diff
@@ -183,21 +150,26 @@ pub fn validate_signal_num(num: c_int, for_vcpu: bool) -> errno::Result<c_int> {
 /// use vmm_sys_util::signal::{register_signal_handler, SignalHandler};
 ///
 /// extern "C" fn handle_signal(_: c_int, _: *mut siginfo_t, _: *mut c_void) {}
-/// unsafe {
-///     register_signal_handler(0, SignalHandler::Siginfo(handle_signal), true, SA_SIGINFO);
-/// }
+/// register_signal_handler(0, handle_signal);
 /// ```
 
-pub unsafe fn register_signal_handler(
-    num: c_int,
-    handler: SignalHandler,
-    for_vcpu: bool,
-    flag: c_int,
-) -> errno::Result<()> {
-    let num = validate_signal_num(num, for_vcpu)?;
-    let mut act: sigaction = handler.into();
-    SignalHandler::set_flags(&mut act, flag);
-    match sigaction(num, &act, null_mut()) {
+pub fn register_signal_handler(num: c_int, handler: SignalHandler) -> errno::Result<()> {
+    validate_signal_num(num)?;
+
+    // Safe, because this is a POD struct.
+    let mut act: sigaction = unsafe { mem::zeroed() };
+    act.sa_sigaction = handler as *const () as usize;
+    act.sa_flags = libc::SA_SIGINFO;
+
+    // Block all signals while the `handler` is running.
+    // This makes async signal operations safe, because
```

Comment:
nit: maybe explain that blocking other signals is needed to make sure the execution of the handler continues uninterrupted if another signal comes